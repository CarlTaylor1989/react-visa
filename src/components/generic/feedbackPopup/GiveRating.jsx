import React from 'react';
import { Translate } from 'react-redux-i18n';
import PropTypes from 'prop-types';
import Rating from '../rating/Rating';
import CancelButton from './CancelButton';

const GiveRating = ({
  currentRating,
  setCurrentRating,
  submitRating,
  closePopup
}) => (
  <>
    <div className="popupTitle">
      <Translate value="generic.feedback.popupTitle" />
    </div>
    <div className="popupText">
      <Translate value="generic.feedback.popupGiveRatingText" />
    </div>
    <Rating currentRating={currentRating} setCurrentRating={setCurrentRating} />
    <div className="buttons">
      <button
        type="button"
        className={`submitBtn${currentRating === 0 ? ' disabled' : ''}`}
        disabled={currentRating === 0}
        onClick={submitRating}
      >
        <Translate className="text" value="generic.feedback.submitBtn" />
      </button>
      <CancelButton onClick={closePopup} />
    </div>
  </>
);

GiveRating.propTypes = {
  currentRating: PropTypes.number.isRequired,
  setCurrentRating: PropTypes.func.isRequired,
  submitRating: PropTypes.func.isRequired,
  closePopup: PropTypes.func.isRequired,
};

export default GiveRating;
