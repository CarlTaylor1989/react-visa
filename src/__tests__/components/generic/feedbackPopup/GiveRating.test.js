/* global toJson */
import React from 'react';
import GiveRating from '../../../../components/generic/feedbackPopup/GiveRating';

let mockProps;
beforeEach(() => {
  jest.resetAllMocks();
  mockProps = {
    currentRating: 0,
    setCurrentRating: jest.fn(),
    submitRating: jest.fn(),
    closePopup: jest.fn()
  };
});

test('renders the GiveRating component', () => {
  const wrapper = shallow(<GiveRating {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('submit button is disabled initially', () => {
  const wrapper = shallow(<GiveRating {...mockProps} />);
  expect(wrapper.find('.submitBtn').hasClass('disabled')).toEqual(true);
});

test('submit button is enabled once currentRating is not zero', () => {
  const wrapper = shallow(<GiveRating {...mockProps} currentRating={4} />);
  expect(wrapper.find('.submitBtn').hasClass('disabled')).toEqual(false);
});

test('rating should be submitted when submit button is clicked', () => {
  // setup
  const wrapper = shallow(<GiveRating {...mockProps} currentRating={3} />);
  wrapper.find('.submitBtn').simulate('click');

  // expectations
  expect(mockProps.submitRating).toHaveBeenCalled();
  expect(mockProps.submitRating).toHaveBeenCalledTimes(1);
});
