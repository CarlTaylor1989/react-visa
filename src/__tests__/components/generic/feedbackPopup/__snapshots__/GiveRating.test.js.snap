// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the GiveRating component 1`] = `
<Fragment>
  <div
    className="popupTitle"
  >
    <Translate
      tag="span"
      value="generic.feedback.popupTitle"
    />
  </div>
  <div
    className="popupText"
  >
    <Translate
      tag="span"
      value="generic.feedback.popupGiveRatingText"
    />
  </div>
  <Rating
    currentRating={0}
    numberOfStars={5}
    setCurrentRating={[MockFunction]}
  />
  <div
    className="buttons"
  >
    <button
      className="submitBtn disabled"
      disabled={true}
      onClick={[MockFunction]}
      type="button"
    >
      <Translate
        className="text"
        tag="span"
        value="generic.feedback.submitBtn"
      />
    </button>
    <CancelButton
      onClick={[MockFunction]}
    />
  </div>
</Fragment>
`;
